lib/google_search_data_viewer/accounts.ex

```diff
@@ -0,0 +1,39 @@
+defmodule GoogleSearchDataViewer.Accounts do
+  import Ecto.Query, warn: false
+
+  alias GoogleSearchDataViewer.Accounts.Passwords
+  alias GoogleSearchDataViewer.Accounts.User
+  alias GoogleSearchDataViewer.Repo
+
+  def list_users, do: Repo.all(User)
+
+  def get_user!(id), do: Repo.get!(User, id)
+
+  def get_user(id), do: Repo.get(User, id)
+
+  def get_user_by_email(email), do: Repo.get_by(User, email: email)
+
+  def create_user(attrs \\ %{}) do
+    %User{}
+    |> User.register_changeset(attrs)
+    |> Repo.insert()
+  end
+
+  def change_user(%User{} = user, attrs \\ %{}), do: User.changeset(user, attrs)
+
+  def validate_email_and_password(email, password) do
+    user = get_user_by_email(email)
+
+    cond do
+      user && Passwords.verify_password(password, user.hashed_password) ->
+        {:ok, user}
+
+      user ->
+        {:error, :unauthorized}
+
+      true ->
+        Bcrypt.no_user_verify()
+        {:error, :not_found}
+    end
```

Comment:
Yeah I seen an example where it should be used if multiple intermediary results need to be true to continue execution, ie multiple true statements in order but I will change this according to your suggestion, thanks

Reply:
One thing I want to bring up in case you may need it in the future. For multiple true statements or generic values (`nil`, `false`) in the `with` block, we usually wrap with the private function and return the atom to identify failure cases. This can make code more readable. 

For example:
```elixir
with :ok <- validate_email(email), 
    :ok <- verify_password(password) do
  # Success case
else
 {:error, :failed_to_validate_email} -> # Handler when failed to validate email
 {:error, :failed_to_verify_password} -> # Handler when failed to verify password
end

defp validate_email(email) do
  case validate_operation do
    true -> :ok
    false -> {:error, :failed_to_validate_email}
  end
end

defp verify_password(password) do
  case verify_operation do
    true -> :ok
    false -> {:error, :failed_to_verify_password}
  end
end
```

Reply:
Noted, thanks